src/operator/every.ts

```diff
@@ -8,38 +8,41 @@ import {Subscriber} from '../Subscriber';
 import {tryCatch} from '../util/tryCatch';
 import {errorObject} from '../util/errorObject';
 
+export type Predicate<T> = (value: T, index: number, source: Observable<T>) => boolean;
+
 /**
  * Returns an Observable that emits whether or not every item of the source satisfies the condition specified.
  * @param {function} predicate a function for determining if an item meets a specified condition.
  * @param {any} [thisArg] optional object to use for `this` in the callback
  * @returns {Observable} an Observable of booleans that determines if all items of the source Observable meet the condition specified.
  */
-export function every<T>(predicate: (value: T, index: number, source: Observable<T>) => boolean,
-                         thisArg?: any): Observable<boolean> {
-  const source = this;
-  if (source._isScalar) {
-    const result: boolean = tryCatch(predicate).call(thisArg || this, source.value, 0, source);
+export function every<T>(predicate: Predicate<T>, thisArg?: any): Observable<boolean> {
+    
+  let _this: Observable<T> = this;
+  if (_this._isScalar) {
+    let scalar: ScalarObservable<T> = <any>_this;
```

Comment:
When this transpiles, won't it have two additional lines where we're setting `_this = this` and then `scalar = _this`? I'm all for good typing for the _exposed_ API. Our internals, though, I just care that they work.